### PR TITLE
fix: adds the qr scanner functionality

### LINKED ISSUE #94: introduce a qr code generator and scanner library
A good library looks like https://pub.dev/packages/qr_flutter

### PR DESCRIPTION
**Pull request by Harshit933**
## Why do we need this

Currently there is no functionality in the app to scan and pay the invoices.

## Changes done

1. Imported the scanner library to scan the invoices.
2. Made a seperate `scanner_view.dart` for scanning purpose.

fix #94

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -47,7 +47,7 @@ android {
         applicationId "com.example.clnapp"
         // You can update the following values to match your application needs.
         // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration.
-        minSdkVersion flutter.minSdkVersion
+        minSdkVersion 21
```
Comment (vincenzopalazzo) [inline android/app/build.gradle:50]: mh why we need this change?

File: lib/views/pay/pay_view.dart [hunk 2]
```diff
@@ -120,6 +148,23 @@ class _PayViewState extends State<PayView> {
     return Scaffold(
       appBar: AppBar(
         elevation: 0,
+        actions: [
+          IconButton(
+            icon: const ImageIcon(AssetImage('assets/images/scanner.png')),
+            onPressed: () async {
+              if (Platform.isAndroid || Platform.isIOS) {
```
Comment (vincenzopalazzo) [inline lib/views/pay/pay_view.dart:155]:
```suggestion
              if (Platform.isAndroid || Platform.isIOS) {
```

no use this to disable the button

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
I did an initial review, but this needs some more work to work with all the platform

File: android/app/build.gradle (hunk 1 above)
Comment (vincenzopalazzo) [reply android/app/build.gradle:50]: we can now increase the flutter.minSdkVersion at this point?

File: lib/views/pay/pay_view.dart [hunk 3]
```diff
@@ -120,6 +137,22 @@ class _PayViewState extends State<PayView> {
     return Scaffold(
       appBar: AppBar(
         elevation: 0,
+        actions: [
+          IconButton(
+            icon: const ImageIcon(AssetImage('assets/images/scanner.png')),
+            onPressed: () async {
+              if (!Platform.isAndroid && !Platform.isIOS) {
+                return;
+              }
```
Comment (vincenzopalazzo) [inline lib/views/pay/pay_view.dart:146]: please hide this on the other platform where this will not work

File: lib/views/pay/pay_view.dart [hunk 4]
```diff
@@ -120,6 +137,22 @@ class _PayViewState extends State<PayView> {
...
+              boltString = await Navigator.of(context).push(MaterialPageRoute(
+                  builder: (context) => ScannerView(
+                        provider: widget.provider,
+                      )));
+              LogManager.getInstance.debug("BOLTSTRING : $boltString");
```
Comment (vincenzopalazzo) [inline lib/views/pay/pay_view.dart:151]:
```suggestion
              LogManager.getInstance.debug("bolt string : `$boltString`");
```

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: lib/views/pay/pay_view.dart [hunk 5]
```diff
@@ -93,11 +110,14 @@ class _PayViewState extends State<PayView> {
...
-              labelText: 'Invoice',
-              hintText: 'invoice',
+              contentPadding: const EdgeInsets.all(30),
+              labelText: 'Invoice/ btc address',
+              hintText: 'Invoice/ btc address',
             ),
           ),
+          const SizedBox(
+            height: 20,
+          ),
```
Comment (vincenzopalazzo) [inline lib/views/pay/pay_view.dart:120]: hey it is not better use `Space` and not a `SizeBox`? imho the Sizebox is an antippattern

**Review by vincenzopalazzo — APPROVED** [review]
Please can you try to fix my comment in a followup PR?

For now this is very good thanks!

File: lib/views/pay/pay_view.dart (hunk 5 above)
Comment (Harshit933) [reply lib/views/pay/pay_view.dart:120]: Sure

Comment (vincenzopalazzo) [reply lib/views/pay/pay_view.dart:120]: I think this will not work hahah sorry

**Pull request merged**